hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -571,4 +572,29 @@ public static int getTargetField(RexProgram calcProgram, int inputFieldIndex) {
         }
         return -1;
     }
+
+    public static int extractKeyParamIndex(final ExpressionValues values, final PartitionedMapTable mapTable) {
```

Comment:
This is specific for INSERT, it should not be in the util class, but in the INSERT rule or rel.